targets/unity-v2/source/Shared/Public/PlayFabExceptions.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Base class for PlayFab exceptions
+    /// </summary>
+    [Serializable]
+    public class PlayFabException : Exception
+    {
+        public PlayFabException() { }
+
+        public PlayFabException(string message)
+            : base(message) { }
+
+        public PlayFabException(string message, Exception inner)
+            : base(message, inner) { }
+
+        protected PlayFabException(SerializationInfo info, StreamingContext context)
+            : base(info, context) { }
+    }
+
+    /// <summary>
+    /// Base class for PlayFab exceptions caused by incorrect settings
+    /// </summary>
+    [Serializable]
+    public class PlayFabSettingsException : PlayFabException
+    {
+        public PlayFabSettingsException(string message)
+            : base(message) { }
+
+        protected PlayFabSettingsException(SerializationInfo info, StreamingContext context)
+            : base(info, context) { }
+    }
+
+    /// <summary>
+    /// Exception that occurs when more than one PlayFabSharedSettings asset is found in resources
+    /// </summary>
+    [Serializable]
+    public class MultipleSharedSettingsException : PlayFabSettingsException
```

Comment:
This one conflicts with a change we're going to make very soon in the SharedSettings object. You were right to include it, but I want to preemptively avoid that conflict.  Please remove this one, and leave the place it's used as a normal exception that we'll fix on our end.

Reply:
Done!